implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -547,4 +553,33 @@ private String extractMetricKey(String metadataLine) {
         return metricKey.toString();
     }
 
+    /**
+     * Maps a unit string to a UCUM-compliant string, if the mapping is known, see:
+     * {@link #ucumTimeUnitMap()}.
+     * @param unit the unit that might be mapped
+     * @return The UCUM-compliant string if known, otherwise returns the original unit
+     */
+    private static String mapUnitIfNeeded(String unit) {
+        return unit != null && UCUM_TIME_UNIT_MAP.containsKey(unit) ? UCUM_TIME_UNIT_MAP.get(unit) : unit;
+    }
+
+    /**
+     * Mapping from OpenJDK's {@link TimeUnit#toString()} and other common time unit
+     * formats to UCUM-compliant format, see: <a href="https://ucum.org/">ucum.org</a>.
+     * @return Time unit mapping to UCUM-compliant format
+     */
+    private static Map<String, String> ucumTimeUnitMap() {
+        Map<String, String> mapping = new HashMap<>();
+        mapping.put("nanoseconds", "ns");
+        mapping.put("nanosecond", "ns");
+        mapping.put("microseconds", "us");
+        mapping.put("microsecond", "us");
+        mapping.put("milliseconds", "ms");
+        mapping.put("millisecond", "ms");
+        mapping.put("seconds", "s");
```

Comment:
If the units are going to be coming from `TimeUnit#toString()`, wouldn't it be safer to build the map using e.g. `TimeUnit.SECONDS.toString()` instead of `"seconds"`?

Reply:
I was thinking the same (got this from @pirgeo) and I even implemented it using `TimeUnit` but then I thought the intention here was to also handle cases when someone manually sets a time unit for a non-time-based `Meter` so I kept this mapping (not a good example because this is already in UCUM format: [`ProcessorMetrics`](https://github.com/micrometer-metrics/micrometer/blob/eff8a6174e422714894ec8857555197493b78ccb/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/ProcessorMetrics.java#L124-L128)).

Reply:
Yes, it would be nice to cover "all" cases. In a perfect world, all data producers would comply with just one standard. That's not very likely to happen anytime soon though, so my attempt was to get started here and at least keep it consistent for these seconds-based units, because these are the ones we have seen make problems in the past. At the end of the day, it shouldn't really matter whether the user sets `milliseconds`, `millisecond` or `ms`, they all refer to the same concept, so I just put them all in to have consistency.
If I recall correctly, using `TimeUnit#toString` yields `milliseconds`, so that is really the most important one to me. I am not opposed to adding them to the map - in fact, that would future-proof us in case the `TimeUnit#toString` ever changes.